Audio Progressive Compression

Samyoul opened the pull request:
First steps to improving audio transport issues, see https://github.com/status-im/status-mobile/issues/13914

Unfortunately there is a bug that I don't really have want to spend more time resolving, it is really taking too much time to resolve. Basically hangs for some encodings params, I don't know why, but it hangs indefinitely and that is a big problem and if I try to interrupt the process I get a segmentation violation crash. And recover isn't seeming to work. Also it would be better if we can decode AAC and then re-encode to AAC, so this whole PR is a massive wash.

## HINT FOR FUTURE DEVELOPERS OF THE FUTURE >>>

One thing to notice, decreasing the `SamplingRate` actually will increase the end file size relative to an encoding with a higher `SamplingRate`, So start with the original `SamplingRate` run through the different `ByteRates` and then increase the `SamplingRate`.

Samyoul committed:
Added some preliminary work for aac compression

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 850d0d77 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2853/1/) | 2022-09-09 14:27:14 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220909-142415-850d0d-pr2853.zip) |
| :heavy_check_mark: | 850d0d77 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2853/1/) | 2022-09-09 14:29:15 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220909-142410-850d0d-pr2853.zip) |
| :heavy_check_mark: | 850d0d77 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2853/1/) | 2022-09-09 14:29:19 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220909-142410-850d0d-pr2853.aar) |
| | | | | | | |
| :heavy_check_mark: | a7aa2e63 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2853/2/) | 2022-09-29 15:19:01 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220929-151638-a7aa2e-pr2853.zip) |
| :heavy_check_mark: | a7aa2e63 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2853/2/) | 2022-09-29 15:21:46 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220929-151638-a7aa2e-pr2853.zip) |
| :heavy_check_mark: | a7aa2e63 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2853/2/) | 2022-09-29 15:22:25 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220929-151638-a7aa2e-pr2853.aar) |

Samyoul committed:
Added some progressive compression

Samyoul committed:
Work on compression from wav to aac

Unfortunately there is a bug that I don't really have want to spend more time resolving, it is really taking too much time to resolve. Basically  hangs for some encodings params, I don't know why, but it hangs indefinitely and that is a big problem and if I try to interupt the process I get a segmentation violation crash. And recover isn't seeming to work. Also it would be better if we can decode AAC and then re-encode to AAC, so this whole PR is a massive wash.

Samyoul commented:
Closing, see the PR description for details.
